### PR TITLE
Generic Enrichment Lookups

### PR DESCRIPTION
**Pull request by JacobMcKenzieSmarty**
Added functionality to the us-enrichment-api to allow for generic lookups that return raw json as a byte slice. This will allow for future data sets to be added without needing to make an immediate change to the SDK.

### CODE REVIEW

File: us-enrichment-api/lookup.go [hunk 1]
```diff
@@ -23,6 +24,45 @@ type enrichmentLookup interface {
...
+	DataSubset string
+	Response   []byte
+}
+
+func (g *genericLookup) getSmartyKey() string     { return g.Lookup.SmartyKey }
+func (g *genericLookup) getDataSet() string       { return g.DataSet }
+func (g *genericLookup) getDataSubset() string    { return g.DataSubset }
+func (g *genericLookup) getLookup() *Lookup       { return g.Lookup }
+func (g *genericLookup) getResponse() interface{} { return g.Response }
+func (g *genericLookup) unmarshalResponse(bytesResponse []byte, headers http.Header) error {
```
Comment (LandonSmarty) [inline us-enrichment-api/lookup.go:39]: To keep consistency with the rest of the lookup structs, lets change the bytesResponse variable name to bytes.

Comment (LandonSmarty) [inline us-enrichment-api/lookup.go:39]: With the logic here, I would love to see a test in client_test.go to verify everything is working properly.

**Review by LandonSmarty — CHANGES_REQUESTED** [review]
The code looks great! I am really glad you simplified the unmarshalResponse function for the generic lookup. Main thing here is just add tests and make the naming consistent with the other structs.

**Review by LandonSmarty — APPROVED** [review]
Everything looks great! Well done.

**Comment by smartybryan** [comment]
The implementation looks good.  
I'm struggling with the name SendGenericLookup.  
I thought about SendLookup but it seems like it needs more. Maybe SendUniversalLookup, or SendGeneralLookup.  Hmm.

**Comment by smartykent** [comment]
I agree with Bryan in both cases.  Some examples would be good.  I also wonder if we can just call it SendLookup.

**Review by SmartyAdam — APPROVED** [review]

**Pull request merged**